tests/server-client-connection.cpp

```diff
@@ -0,0 +1,114 @@
+/**
+ * Unit tests
+ */
+
+#include <gtest/gtest.h>
+
+#include "client/server-connection.hpp"
+#include "server-stats.hpp"
+#include "server/client-connection.hpp"
+#include "server/server.hpp"
+#include "tec-types.hpp"
+#include <asio.hpp>
+#include <file-factories.hpp>
+#include <spdlog/sinks/stdout_sinks.h>
+#include <spdlog/spdlog.h>
+
+namespace tec {
+namespace networking {
+
+class ClientCommandReceptor : public EventQueue<ClientCommandsEvent> {
+public:
+	ClientCommandReceptor() { last_event = promised_event.get_future(); }
+	void process() { EventQueue<ClientCommandsEvent>::ProcessEventQueue(); }
+	using EventQueue<ClientCommandsEvent>::On;
+	virtual void On(std::shared_ptr<ClientCommandsEvent> data) { promised_event.set_value(data); }
```

Comment:
Should this be override instead of virtual?

Reply:
my finest copy paste hacking. I probably should add override, having virtual could go either way, but since it doesn't matter I could probably just remove the virtual.